feat: Improve faro.receiver.sourcemaps caching strategy (#4337)
grafana/alloy: docs/sources/reference/components/faro/faro.receiver.md

The `sourcemaps` block configures how to retrieve sourcemaps.
Sourcemaps are then used to transform file and line information from minified code into the file and line information from the original source code.

> | Name                               | Type           | Description                                                                       | Default | Required |
> |------------------------------------|----------------|-----------------------------------------------------------------------------------|---------|----------|
> | `download`                         | `bool`         | Whether to download sourcemaps.                                                   | `true`  | no       |
> | `download_from_origins`            | `list(string)` | Which origins to download sourcemaps from.                                        | `["*"]` | no       |
> | `download_timeout`                 | `duration`     | Timeout when downloading sourcemaps.                                              | `"1s"`  | no       |
> | `cache_minimum_ttl`                | `duration`     | Duration after which source map is deleted from cache if not used                 | `inf`   | no       |
> | `cache_error_cleanup_interval`     | `duration`     | Duration after which the download of source map that previously failed is retried | `"1h"`  | no       |
> | `cache_cleanup_check_interval`     | `duration`     | How often should cached sourcemaps be checked for cleanup                         | `"1h"`  | no       |

Reviewer: ```suggestion
| Name                           | Type           | Description                                                                       | Default | Required |
| ------------------------------ | -------------- | --------------------------------------------------------------------------------- | ------- | -------- |
| `cache_cleanup_check_interval` | `duration`     | How often should cached sourcemaps be checked for cleanup                         | `"1h"`  | no       |
| `cache_error_cleanup_interval` | `duration`     | Duration after which the download of source map that previously failed is retried | `"1h"`  | no       |
| `cache_minimum_ttl`            | `duration`     | Duration after which source map is deleted from cache if not used                 | `inf`   | no       |
| `download_from_origins`        | `list(string)` | Which origins to download sourcemaps from.                                        | `["*"]` | no       |
| `download_timeout`             | `duration`     | Timeout when downloading sourcemaps.                                              | `"1s"`  | no       |
| `download`                     | `bool`         | Whether to download sourcemaps.                                                   | `true`  | no       |
```

Just a small tidy on the table and alpha sort rows

By default, sourcemap downloads are subject to a timeout of `"1s"`, specified by the `download_timeout` argument.
Setting `download_timeout` to `"0s"` disables timeouts.

> By default, sourcemaps are held in memory indefinitely. By setting the `cache_minimum_ttl` sourcemaps will be cleared if not used during the specified duration.
> 
> By default, if there is an error while downloading or parsing a sourcemap, error gets cached. After duration specified by `cache_error_cleanup_interval`, all errors get cleared from cache.
> 
> By default, every 30s cached sourcemaps are checked for cleanup. The frequency of checking can be modified by setting the `cache_cleanup_check_interval` argument.

Reviewer: ```suggestion
By default, sourcemaps are held in memory indefinitely.
You can set `cache_minimum_ttl` to clear sourcemaps that aren't used during the specified duration.

By default, if there's an error while downloading or parsing a sourcemap, the error is cached.
After the duration specified by `cache_error_cleanup_interval`, all errors are cleared from the cache.

By default, cached sourcemaps are checked for cleanup every 30 seconds.
You can modify the frequency by setting the `cache_cleanup_check_interval` argument.
```

To retrieve sourcemaps from disk instead of the network, specify one or more [`location` blocks][location].
When `location` blocks are provided, they're checked first for sourcemaps before falling back to downloading.

#### `cache`

The `cache` block configures sourcemap caching behavior.

> All cache settings are optional with sensible defaults.

Reviewer: ```suggestion
The `cache` block configures sourcemap caching behavior.
```

I don't think we really need that second sentence. We have many blocks with optional args that have default values.

| Name                     | Type       | Description                                                                        | Default | Required |
| ------------------------ | ---------- | ---------------------------------------------------------------------------------- | ------- | -------- |
| `cleanup_check_interval` | `duration` | How often cached sourcemaps are checked for cleanup.                               | `"30s"` | no       |
| `error_cleanup_interval` | `duration` | Duration after which the download of source map that previously failed is retried. | `"1h"`  | no       |

> | `ttl`                    | `duration` | Duration after which source map is deleted from cache if not used.                 | `inf`   | no       |

Reviewer: ```suggestion
| Name                     | Type       | Description                                                                               | Default | Required |
| ------------------------ | ---------- | ----------------------------------------------------------------------------------------- | ------- | -------- |
| `cleanup_check_interval` | `duration` | How often {{< param "PRODUCT_NAME" >}} checks cached `sourcemaps` for cleanup.            | `"30s"` | no       |
| `error_cleanup_interval` | `duration` | How long {{< param "PRODUCT_NAME" >}} waits before retrying a failed source map download. | `"1h"`  | no       |
| `ttl`                    | `duration` | How long {{< param "PRODUCT_NAME" >}} keeps an unused source map in the cache.            | `inf`   | no       |
```
